src/impl/vamp/random/rng.hh

```diff
@@ -0,0 +1,15 @@
+#pragma once
+
+#include <memory>
+#include <vamp/vector.hh>
+
+namespace vamp::rng
+{
+    template <std::size_t dim>
+    struct RNG
+    {
+        using Ptr = std::shared_ptr<RNG<dim>>;
+        virtual inline void reset() noexcept = 0;
+        virtual inline auto next() noexcept -> FloatVector<dim> = 0;
```

Comment:
Thoughts on exposing `seed` as well? For Halton, this is skipping iterations; for XORShift, this reinitializes with the given seed.

Reply:
Maybe for later - exposing that is a small chore that we can do if someone needs it.